### PR TITLE
Automatically pick-up all listeners

### PR DESCRIPTION
**Pull request by tuhrig**

### CODE REVIEW

File: src/main/kotlin/de/bringmeister/spring/aws/kinesis/KinesisListenerStarter.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * This class will automatically start/register all available Kinesis listeners.
+ * As soon as the Spring Boot application has started, this class will iterate
+ * through the list of all available listeners and start them. This is a very
+ * convenient function for the developer using the library. He must only define
+ * the listener - the rest will be managed by the library.
+ */
+class KinesisListenerStarter(private val inboundGateway: AwsKinesisInboundGateway,
+                             private val kinesisListeners: MutableList<KinesisListener<*, *>>) {
+
+    @EventListener(ContextRefreshedEvent::class)
```
Comment (tuhrig) [inline src/main/kotlin/de/bringmeister/spring/aws/kinesis/KinesisListenerStarter.kt]: Verify event!

File: src/main/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisInboundGateway.kt [hunk 2]
```diff
@@ -16,10 +16,10 @@ class AwsKinesisInboundGateway(private val clientProvider: AwsKinesisClientProvi
         register(KinesisListener.build(streamName, eventHandler, dataClass, metadataClass))
     }
 
-    fun <D, M> register(handler: KinesisListener<D, M>) {
+    fun register(handler: KinesisListener<*, *>) {
```
Comment (tuhrig) [inline src/main/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisInboundGateway.kt]: Revert

File: src/main/kotlin/de/bringmeister/spring/aws/kinesis/KinesisListenerStarter.kt (hunk 1 above)
Comment (jmatusewicz-bringmeister) [reply src/main/kotlin/de/bringmeister/spring/aws/kinesis/KinesisListenerStarter.kt]: I would rather listen for `ApplicationReadyEvent`

Comment (tuhrig) [reply src/main/kotlin/de/bringmeister/spring/aws/kinesis/KinesisListenerStarter.kt]: OK

**Comment by jmatusewicz-bringmeister** [comment]
👍

**Pull request merged**